docs/aws-iam-setup.md

```diff
@@ -0,0 +1,120 @@
+
+To configure AWS IAM authentication means to set up your AWS environment and configure Percona Server for MongoDB. The AWS environment setup is out of scope of this document. Consult the AWS documentation to perform the following setup steps: 
+
+1.	[Configure the AWS service to work with IAM and support temporary credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_aws-services-that-work-with-iam.html)
+2.	For user authentication: 
+      
+      * [Create the IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html) and copy its ARN (Amazon Resource Name)
+
+    For role authentication:
+
+      * [Create the IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-service.html)
+      * Attach the IAM role to the AWS service.
+      * Copy the ARN of the IAM role. 
+
+## Configure Percona Server for MongoDB
+
+The steps are the following:
+
+1. Add records for the IAM user/role in the `$external` database
+2. Enable external authentication
+
+### Add IAM users / roles
+
+During the authentication, Percona Server for MongoDB matches the ARN of the IAM user or role retrieved from AWS STS against the user created in the `$external` database. Thus, the username for this user must include their ARN  and have the following format:
+
+=== "User authentication"
+
+     ```
+     arn:aws:iam::<ARN>:user/<user_name>
+     ```
+
+=== "Role authentication"
+
+     ```
+     arn:aws:iam::<ARN>:role/<role_name>
+     ```
+
+Create a user and assign the required roles to them. Specify the ARN and names in the following example commands:
+
+=== "User authentication"
+
+     ```{.javascript data-prompt=">"}
+     > use $external
+     > db.createUser(
+     	{
+     		user: "arn:aws:iam::000000000000:user/myUser",
+     		roles: [{role: "read", db: "admin"}]
+     	}
+     )
+     ```
+
+=== "Role authentication"
+
+     ```{.javascript data-prompt=">"}
+     > use $external
+     > db.createUser(
+     	{
+     		user: "arn:aws:iam::111111111111:role/myRole",
+     		roles: [{role: "read", db: "admin"}]
+     	}
+     )
+     ```
+
+### Enable external authentication
+
+Run the following commands as root or via `sudo`
+
+1. Stop the `mongod` service
+
+    ```{.bash data-prompt="$"}
+    $ sudo systemctl stop mongod
+    ```
+
+2. Edit the `/etc/mongod.conf` configuration file
+
+    ```yaml
+    security:
+      authorization: enabled
+
+    setParameter:
+      authenticationMechanisms: MONGODB-AWS
+    ```
+
+3. Start the `mongod` service
+
+    ```{.bash data-prompt="$"}
+    $ sudo systemctl start mongod
+    ```
+
+## Authenticate in Percona Server for MongoDB using AWS IAM
+
+To test the authentication, run the following command to connect to Percona Server for MongoDB
```

Comment:
This will only work if you run this command on the machine running in one of the Amazon's cloud services. Otherwise you will need to explicitly specify UserKey and SecretKey.

Reply:
Let's provide all possible ways then:
1. MONGODB URI
mongo 'mongodb://<aws access key id>:<aws secret access key>@IP_or_hostname/testdb?authSource=$external&authMechanism=MONGODB-AWS'
OR
mongo 'mongodb://<aws access key id>:<aws secret access key>@IP_or_hostname/testdb?authSource=$external&authMechanism=MONGODB-AWS&authMechanismProperties=AWS_SESSION_TOKEN:<aws session token>'

2. Environment variables
a) user-defined
export AWS_ACCESS_KEY_ID='<aws access key id>'
export AWS_SECRET_ACCESS_KEY='<aws secret access key>'
export AWS_SESSION_TOKEN='<aws session token>' 

b) if mongo client runs on AWS resources like EC2 or ECS which have IAM role assigned, client  gets credentials from preset environment variables

mongo --authenticationMechanism=MONGODB-AWS --authenticationDatabase='$external'

Reply:
Added